json-path/src/main/java/com/jayway/jsonpath/spi/json/JacksonJsonNodeJsonProvider.java

```diff
@@ -153,6 +154,22 @@ public Object getMapValue(Object obj, String key) {
         }
     }
 
+    @Override
+    public Object getProperty(Object obj, Object key) {
```

Comment:
Make these changes in the abstract class, as these are common, all the changes you make should benefit all providers.